code/game/turfs/simulated/floor/asteroid.dm

```diff
@@ -0,0 +1,317 @@
+	..()
+	name = proper_name
+	if(prob(floor_variance))
+		icon_state = "[environment_type][rand(0,12)]"
+
+/turf/simulated/floor/plating/asteroid/burn_tile()
+	return
+
+/turf/simulated/floor/plating/asteroid/ex_act(severity, target)
+	switch(severity)
+		if(3.0)
+			return
+		if(2.0)
+			if(prob(20))
+				gets_dug()
+		if(1.0)
+			gets_dug()
+
+/turf/simulated/floor/plating/asteroid/attackby(obj/item/W, mob/user, params)
+	//note that this proc does not call ..()
+	if(!W || !user)
+		return 0
+
+	if((istype(W, /obj/item/shovel) || istype(W, /obj/item/pickaxe)))
+		var/turf/T = get_turf(user)
+		if(!istype(T))
+			return
+
+		if(dug)
+			to_chat(user, "<span class='warning'>This area has already been dug!</span>")
+			return
+
+		to_chat(user, "<span class='notice'>You start digging...</span>")
+		playsound(src, W.usesound, 50, 1)
+		if(do_after(user, 20 * W.toolspeed, target = src))
+			to_chat(user, "<span class='notice'>You dig a hole.</span>")
+			gets_dug()
+			return
+
+	if(istype(W,/obj/item/storage/bag/ore))
+		var/obj/item/storage/bag/ore/S = W
+		if(S.collection_mode == 1)
+			for(var/obj/item/stack/ore/O in src.contents)
+				O.attackby(W,user)
+				return
+
+	if(istype(W, /obj/item/stack/tile))
+		var/obj/item/stack/tile/Z = W
+		if(!Z.use(1))
+			return
+		ChangeTurf(Z.turf_type)
+		playsound(src, 'sound/weapons/Genhit.ogg', 50, 1)
+
+/turf/simulated/floor/plating/asteroid/gets_drilled()
+	if(!dug)
+		gets_dug()
+	else
+		..()
+
+/turf/simulated/floor/plating/asteroid/proc/gets_dug()
+	if(dug)
+		return
+	for(var/i in 1 to 5)
+		new sand_type(src)
+	dug = 1
+	icon_plating = "[environment_type]_dug"
+	icon_state = "[environment_type]_dug"
+	slowdown = 0
+	return
+
+/turf/simulated/floor/plating/asteroid/singularity_act()
+	return
+
+/turf/simulated/floor/plating/asteroid/singularity_pull(S, current_size)
+	return
+
+/turf/simulated/floor/plating/asteroid/basalt
+	name = "volcanic floor"
+	baseturf = /turf/simulated/floor/plating/asteroid/basalt
+	icon_state = "basalt"
+	icon_plating = "basalt"
+	environment_type = "basalt"
+	sand_type = /obj/item/stack/ore/glass/basalt
+	floor_variance = 15
+
+/turf/simulated/floor/plating/asteroid/basalt/lava //lava underneath
+	baseturf = /turf/simulated/floor/plating/lava/smooth
+
+/turf/simulated/floor/plating/asteroid/basalt/airless
+	temperature = TCMB
+	oxygen = 0
+	nitrogen = 0
+
+/turf/simulated/floor/plating/asteroid/basalt/New()
+	..()
+	switch(icon_state)
+		if("basalt1", "basalt2", "basalt3") //5 and 9 are too dark to glow and make the amount of glows in tunnels too high
+			set_light(1, 1) //this is basically a 3.75% chance that a basalt floor glows
+
+/turf/simulated/floor/plating/asteroid/basalt/gets_dug()
+	if(!dug)
+		set_light(0)
+	..()
+
+///////Surface. The surface is warm, but survivable without a suit. Internals are required. The floors break to chasms, which drop you into the underground.
+
+/turf/simulated/floor/plating/asteroid/basalt/lava_land_surface
+	oxygen = 14
```

Comment:
Probably need to turn these values into defines. Also, I'm not sure if this is the correct mix as /tg/ uses initial_gas_mix.

Reply:
This is the correct mix.